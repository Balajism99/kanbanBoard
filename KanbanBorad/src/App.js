import React  from "react";

import Kanban from "./components/kanban";
import FullScreenDialog from "./components/PopUp";
import PDF from "./components/ss";
import Render from "./components/ss";
import SS from "./components/ss";



export default function App() {
 return(
   <div>
     <Kanban />
     <FullScreenDialog />
   </div>
 )
}














/*



 var string = doc.output('datauristring');
 var embed = "<embed width='100%' height='100%' src='" + string + "'/>"
 document.open();
 document.write(embed);
 document.close();






https://makitweb.com/take-screenshot-of-webpage-with-html2canvas/






<Snackbar className="bar" anchorOrigin={{vertical:"top", horizontal:"center"}} open={barOpen} onClose={barClose}   autoHideDuration={4000} >
<Alert severity="error" className="message">{errors}
<IconButton key="close" aria-label="Close" color="inherit" onClick={barClose} autoHideDuration={6000}>X</IconButton>
   <button className="btn btn-primary" onClick={generatePdf}>Preview</button>
  
</Alert>
   </Snackbar>

   */
//
/*pdf.save("mypdf.pdf")
  
  function generatePdf(){
    var doc = new jsPDF("l", "pt", "a1")
    doc.html(document.querySelector("#content"),{
      callback : function(pdf){
        
        var string = doc.output('datauristring');
        var embed = "<embed width='100%' height='100%' src='" + string + "'/>"
        var preview = window.open();
        preview.document.open();
        preview.document.write(embed);
        preview.document.close();
      }
    })
  }

   
<div>
      <h1  className="error"><u>{errors}</u></h1>
    <div className="main-comp">
      <DragDropContext
        onDragEnd={result => onDragEnd(result,  columns, setColumns,setErrors)}
      >
       
        {Object.entries(columns).map(([columnId, column], index) => {
          return (
    
            <div className="main-col" key={columnId} 
            >
              <h2>{column.name} </h2>
              <div className="col">
                <Droppable droppableId={columnId} key={columnId}>
                  {(provided, snapshot) => {
                    return (
                      
                      <div className="col-1"
                        {...provided.droppableProps}
                        ref={provided.innerRef}
                        style={{
                          background: snapshot.isDraggingOver
                            ? "lightblue"
                            : "lightgrey",
                        }}
                      >
                        {column.items.map((item, index) => {
                          return (
                            <Draggable
                              key={item.id}
                              draggableId={item.id}
                              index={index}
                            >
                              
                              {(provided, snapshot) => {
                                return (
                                  <div
                                  className="col-2"
                                    ref={provided.innerRef}
                                    {...provided.draggableProps}
                                    {...provided.dragHandleProps}
                                    style={{
                                      background: column.color,
                                      ...provided.draggableProps.style
                                    }}
                                  >
                                   Name: {item.content} {column.icon} <br></br>
                                   Reg-No: {item.regNo}<br></br>
                                   Dept: {item.dept}
                                  </div>
                                );
                              }}
                            </Draggable>
                          );
                        })}
                        {provided.placeholder}
                      </div>
                     
                    );
                  
                  }}
                </Droppable>
              </div>
            </div>
            
          );
        })}
      </DragDropContext>
    </div>
   <button className="preview-btn" onClick={generatePdf}>Preview</button>
    </div>

    
<div>
  {errors ?(
   <Snackbar className="bar" severity="error" anchorOrigin={{vertical:"top", horizontal:"center"}} open={barOpen} onClose={barClose} color="red"  autoHideDuration={4000} 
   message={<span  className="message">{errors}</span>}
   action={[
     <IconButton key="close" aria-label="Close" color="inherit" onClick={barClose} autoHideDuration={6000}>X</IconButton>
   
   ]}   >
   </Snackbar>
  ):
  (
    <span></span>
  )
}

    <div className="main-comp" id="content">
   
      <DragDropContext
        onDragEnd={result => onDragEnd(result,  columns, setColumns,setErrors)}
      >
       
        {Object.entries(columns).map(([columnId, column], index) => {
          return (
    
            <Container key={columnId} 
            >
              <h2>{column.name} </h2>
              <Row>
                <Droppable droppableId={columnId} key={columnId}>
                  {(provided, snapshot) => {
                    return (
                      
                      <Col auto className="col-1"
                        {...provided.droppableProps}
                        ref={provided.innerRef}
                        style={{
                          background: snapshot.isDraggingOver
                            ? "lightblue"
                            : "lightgrey",
                        }}
                      >
                        {column.items.map((item, index) => {
                          return (
                            <Draggable
                              key={item.id}
                              draggableId={item.id}
                              index={index}
                            >
                              
                              {(provided, snapshot) => {
                                return (
                                  <Col auto   className="col-2"
                                    ref={provided.innerRef}
                                    {...provided.draggableProps}
                                    {...provided.dragHandleProps}
                                    style={{
                                      background: column.color,
                                      ...provided.draggableProps.style
                                    }}
                                  >
                                   Name: {item.content} {column.icon} <br></br>
                                   Reg-No: {item.regNo}<br></br>
                                   Dept: {item.dept}
                                  </Col>
                                );
                              }}
                            </Draggable>
                          );
                        })}
                        {provided.placeholder}
                      </Col>
                     
                    );
                  
                  }}
                </Droppable>
              </Row>
            </Container>
            
          );
        })}
      </DragDropContext>
    </div>
    <button className="preview-btn" onClick={generatePdf}>Preview</button>
    </div> */