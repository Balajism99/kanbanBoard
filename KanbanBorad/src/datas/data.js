import  {v4 as uuid} from "uuid";


 export const datas = [
    { id: uuid(), content: "Harish" , regNo:154796, dept:"CSC"},
    { id: uuid(), content: "Pranesh",regNo:458136, dept:"IT" },
    { id: uuid(), content: "Ajay",regNo:124769, dept:"IT" },
    { id: uuid(), content: "Karthick",regNo:238756, dept:"ECE" },
    { id: uuid(), content: "Vijay",regNo:125796, dept:"EEE" }
  ];