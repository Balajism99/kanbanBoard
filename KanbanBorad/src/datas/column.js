import  {v4 as uuid} from "uuid";
import { datas} from "./data";



export const columnView = {
    [uuid()]: {
      name: "Screening",
      items: datas,
      //icon: "⭕️",
      color:"rgb(88, 88, 175)" ,
      no:1
    },
    [uuid()]: {
      name: "Written Test",
      items: [],
      //icon: "✍🏻",
      color:"rgb(155, 146, 72)",
      no:2
    },
    [uuid()]: {
      name: " HR",
      items: [],
      //icon: "⏳",
      color:"black",
      no:3
    },
    [uuid()]: {
      name: "Selected",
      items: [],
     // icon: "✅",
      color:"green",
      no:4
    },
    [uuid()]: {
      name: "Offer Letter ",
      items: [],
     // icon: "📝",
      color:"rgb(32, 114, 180)",
      no:5
    }
  };