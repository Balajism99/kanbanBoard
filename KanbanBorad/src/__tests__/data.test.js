import { fireEvent, render } from "@testing-library/react";
import App from "../App";
import Header from "../Headers";
import { renderHook } from "@testing-library/react-hooks";
import { act, createRenderer } from "react-dom/test-utils";
import { columnView } from "../datas/column";
import { useCallback, useState } from "react";
import PDF from "../preview/toPdf";
import renderer from "react-test-renderer"


/*test("by data test id", () =>{
    const input=render(<Header />)
const element = screen.getAllByTestId('board')

})*/

it("renders main app",()=>{
    render(<App />)
})

it("renders main header",()=>{
    render(<Header />)
})


describe("hook",()=>{
it('renders default text', () => {
  const { container } = render(<App/>)
container.querySelector('Name')
});
})


describe("custom hook tests", ()=>{
    function validateError(){
        const [error, setError]=useState('')
        
    const valid =" CANDIDATE MUST GO THROUGH EACH AND EVERY LEVELS, CANNOT SKIP AHEAD ";
    const validBack =" CANNOT DRAG BACKWARDS ";

    const validerr =useCallback(() => setError(valid));
    const validBackerr =useCallback(() => setError(validBack));
    return{error, validerr, validBackerr}
    }
    test('should update front error', () => {
        const { result } = renderHook(() => validateError())
  
        act(() => result.current.validerr())
  
        expect(result.current.error).toBe(" CANDIDATE MUST GO THROUGH EACH AND EVERY LEVELS, CANNOT SKIP AHEAD ")
      });

      test('should update front error', () => {
        const { result } = renderHook(() => validateError())
  
        act(() => result.current.validBackerr())
  
        expect(result.current.error).toBe(" CANNOT DRAG BACKWARDS ")
      })
})
describe("custom hook tests 2", ()=>{
    function updateColumn(){
        const [column, setColumn]=useState(columnView)
    const valid =column
    const upd= useCallback(()=> setColumn(valid))
    return{column, upd}
        }
        
    test('should update hook', () => {
        const { result } = renderHook(() => updateColumn())
  
        act(() => result.current.upd())
  
        expect(result.current.column)
      })
})

it("renders pdf",()=>{
    render(<PDF />)
})


test("should be able to preview pdf", () =>{
    const mockFn = jest.fn();
    const {getByTestId} =render(<PDF handleSubmit={mockFn} />);
    const button = getByTestId("btn");
    fireEvent.submit(button);

    expect(mockFn).toHaveBeenCalledTimes(0);
})

test("function testing",()=>{
    function generatePreview(){
        var doc = new jsPDF("l", "pt", "a1")
        doc.html(document.querySelector("#content"),{
          callback : function(pdf){
            pdf.save("mypdf.pdf")
          },
        })
      }
    expect(generatePreview)
})

it ("matches snapshot", ()=>{
const tree= renderer.create(<App />);
expect(tree).toMatchSnapshot();
})