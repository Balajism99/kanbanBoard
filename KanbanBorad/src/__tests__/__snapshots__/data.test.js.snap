// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <span />
  <div
    className="main-comp"
    id="content"
  >
    <div
      className="container"
    >
      <h2>
        Screening
         
      </h2>
      <div
        className="row"
      >
        <div
          auto={true}
          className="col-1 col"
          data-rbd-droppable-context-id="4"
          data-rbd-droppable-id="0b6e0eb6-29a6-4c1c-817f-0e903fca75fd"
          style={
            Object {
              "background": "lightgrey",
            }
          }
        >
          <div
            aria-describedby="rbd-hidden-text-4-hidden-text-44"
            auto={true}
            className="col-2 col"
            data-rbd-drag-handle-context-id="4"
            data-rbd-drag-handle-draggable-id="429f6041-fc1c-4190-a1af-7877d5d7b4d3"
            data-rbd-draggable-context-id="4"
            data-rbd-draggable-id="429f6041-fc1c-4190-a1af-7877d5d7b4d3"
            draggable={false}
            onDragStart={[Function]}
            onTransitionEnd={null}
            role="button"
            style={
              Object {
                "background": "rgb(88, 88, 175)",
                "transform": null,
                "transition": null,
              }
            }
            tabIndex={0}
          >
            Name: 
            Harish
            <br />
            Reg-No: 
            154796
            <br />
            Dept: 
            CSC
          </div>
          <div
            aria-describedby="rbd-hidden-text-4-hidden-text-44"
            auto={true}
            className="col-2 col"
            data-rbd-drag-handle-context-id="4"
            data-rbd-drag-handle-draggable-id="8e1bf1f9-571d-4634-8a02-a9823d19f723"
            data-rbd-draggable-context-id="4"
            data-rbd-draggable-id="8e1bf1f9-571d-4634-8a02-a9823d19f723"
            draggable={false}
            onDragStart={[Function]}
            onTransitionEnd={null}
            role="button"
            style={
              Object {
                "background": "rgb(88, 88, 175)",
                "transform": null,
                "transition": null,
              }
            }
            tabIndex={0}
          >
            Name: 
            Pranesh
            <br />
            Reg-No: 
            458136
            <br />
            Dept: 
            IT
          </div>
          <div
            aria-describedby="rbd-hidden-text-4-hidden-text-44"
            auto={true}
            className="col-2 col"
            data-rbd-drag-handle-context-id="4"
            data-rbd-drag-handle-draggable-id="33857c74-0e89-48b5-8206-d9972df53e68"
            data-rbd-draggable-context-id="4"
            data-rbd-draggable-id="33857c74-0e89-48b5-8206-d9972df53e68"
            draggable={false}
            onDragStart={[Function]}
            onTransitionEnd={null}
            role="button"
            style={
              Object {
                "background": "rgb(88, 88, 175)",
                "transform": null,
                "transition": null,
              }
            }
            tabIndex={0}
          >
            Name: 
            Ajay
            <br />
            Reg-No: 
            124769
            <br />
            Dept: 
            IT
          </div>
          <div
            aria-describedby="rbd-hidden-text-4-hidden-text-44"
            auto={true}
            className="col-2 col"
            data-rbd-drag-handle-context-id="4"
            data-rbd-drag-handle-draggable-id="7dde82e4-d3d5-4b3b-b0ff-0be0a836ee97"
            data-rbd-draggable-context-id="4"
            data-rbd-draggable-id="7dde82e4-d3d5-4b3b-b0ff-0be0a836ee97"
            draggable={false}
            onDragStart={[Function]}
            onTransitionEnd={null}
            role="button"
            style={
              Object {
                "background": "rgb(88, 88, 175)",
                "transform": null,
                "transition": null,
              }
            }
            tabIndex={0}
          >
            Name: 
            Karthick
            <br />
            Reg-No: 
            238756
            <br />
            Dept: 
            ECE
          </div>
          <div
            aria-describedby="rbd-hidden-text-4-hidden-text-44"
            auto={true}
            className="col-2 col"
            data-rbd-drag-handle-context-id="4"
            data-rbd-drag-handle-draggable-id="2d26cb27-82d5-423b-a84e-e57dc5f675fb"
            data-rbd-draggable-context-id="4"
            data-rbd-draggable-id="2d26cb27-82d5-423b-a84e-e57dc5f675fb"
            draggable={false}
            onDragStart={[Function]}
            onTransitionEnd={null}
            role="button"
            style={
              Object {
                "background": "rgb(88, 88, 175)",
                "transform": null,
                "transition": null,
              }
            }
            tabIndex={0}
          >
            Name: 
            Vijay
            <br />
            Reg-No: 
            125796
            <br />
            Dept: 
            EEE
          </div>
        </div>
      </div>
    </div>
    <div
      className="container"
    >
      <h2>
        Written Test
         
      </h2>
      <div
        className="row"
      >
        <div
          auto={true}
          className="col-1 col"
          data-rbd-droppable-context-id="4"
          data-rbd-droppable-id="7dbbec61-c2bd-4bd2-82cb-63f85b0cbc76"
          style={
            Object {
              "background": "lightgrey",
            }
          }
        />
      </div>
    </div>
    <div
      className="container"
    >
      <h2>
         HR
         
      </h2>
      <div
        className="row"
      >
        <div
          auto={true}
          className="col-1 col"
          data-rbd-droppable-context-id="4"
          data-rbd-droppable-id="b9b0ae76-f491-4b51-9aa6-76c12dbe0ba8"
          style={
            Object {
              "background": "lightgrey",
            }
          }
        />
      </div>
    </div>
    <div
      className="container"
    >
      <h2>
        Selected
         
      </h2>
      <div
        className="row"
      >
        <div
          auto={true}
          className="col-1 col"
          data-rbd-droppable-context-id="4"
          data-rbd-droppable-id="72d47902-676d-40a5-9193-b5c885279424"
          style={
            Object {
              "background": "lightgrey",
            }
          }
        />
      </div>
    </div>
    <div
      className="container"
    >
      <h2>
        Offer Letter 
         
      </h2>
      <div
        className="row"
      >
        <div
          auto={true}
          className="col-1 col"
          data-rbd-droppable-context-id="4"
          data-rbd-droppable-id="21356831-c3b7-49fa-88df-2bab5bd63392"
          style={
            Object {
              "background": "lightgrey",
            }
          }
        />
      </div>
    </div>
  </div>
</div>
`;
